src/main/java/uk/gov/hmcts/darts/common/entity/AnnotationDocumentEntity.java

```diff
@@ -24,7 +25,7 @@
 @Getter
 @Table(name = "annotation_document")
 public class AnnotationDocumentEntity extends ModifiedBaseEntity
-    implements ConfidenceAware, SoftDelete {
+    implements ConfidenceAware, SoftDelete, HasIntegerId {
```

Comment:
Although probably seen as out of scope for this ticket, we should apply this interface to either all entities or specifically to the softdelete objects (by SoftDelete extending HasIntegerId)

Reply:
SoftDelete does not rely on HasIntegerId so I am reluctant to combine these as downstream we could have a SoftDelete with a String id, etc.

Regarding HasIntegerId we only need to add this interface to classes that benefit from the abstraction